backend/controller/userController.js

```diff
@@ -49,3 +52,74 @@ exports.getExpertInfo = catchAsync(async (req, res, next) => {
     },
   });
 });
+
+exports.sendResetPasswordEmail = catchAsync(async (req, res, next) => {
+  if (!req.body.email) {
+    return next(
+      new AppError("Please provide email to reset your password", 400)
+    );
+  }
+
+  // Find user by email
+  const user = await User.findOne({ email: req.body.email });
+  if (!user) {
+    return next(new AppError("Could not found any user, which is associated with this email.", 400));
+  }
+
+  // Create token
+  const token = crypto.randomBytes(32).toString("hex");
+
+  // Sent a reset password to user email
+  const link = `${process.env.BASE_URL}/user/reset-password?id=${user._id}&token=${token}`;
+  await sendEmail(user.email, link);
+
+  // After sending email success, we save the token
+  await user.update({
```

Comment:
I got a deprecation warning from mongoose today. Can you change `user.update` to this to avoid this warning. 
```suggestion
  await User.findByIdAndUpdate(user.id, {
    resetPasswordToken: token,
    expiredTokenAt: Date.now() + 1000 * 60 * 15, // 15 min expiry
  });
```

![deprecation-warning](https://user-images.githubusercontent.com/43143636/131961544-da035a7c-2bab-41ae-9dd8-84113c2703bb.png)